src/frontend/github-oauth.ts

```diff
@@ -148,14 +151,39 @@ export default (opts: OAuthOptions): GithubOAuth => {
 		checkGithubAuth: async (req: Request, res: Response, next: NextFunction) => {
 			const traceLogsEnabled = await booleanFlag(BooleanFlags.TRACE_LOGGING, false);
 			const tracer = new Tracer(logger.child(opts), "checkGithubAuth", traceLogsEnabled);
+			try {
+				const { githubToken } = req.session;
+				if (!githubToken) {
+					tracer.trace("github token missing, calling login()");
+					throw "Missing github token";
+				}
+				tracer.trace("found github token in session. validating token with API.");
 
-			if (!req.session.githubToken) {
-				tracer.trace("found github token in session, calling login()");
-				res.locals.redirect = req.originalUrl;
-				return login(req, res);
+				await axios.get(`https://api.${envVars.GITHUB_DOMAIN}`, {
+					headers: {
+						Authorization: `Bearer ${githubToken}`
+					}
+				});
+
+				req.log.debug(`Github token is valid, continuing...`);
+
+				// Everything's good, set it to res.locals
+				res.locals.githubToken = githubToken;
+				// TODO: Not a great place to put this, but it'll do for now
+				res.locals.github = GithubApi({ auth: githubToken });
```

Comment:
If the token is good, we replace the Github API local with one including the githubToken for auth.